users/vieting/experiments/librispeech/librispeech_100_ctc/fairseq_finetuning/ctc_standalone/serializer.py

```diff
@@ -0,0 +1,130 @@
+"""
+Contains the serializers for the ReturnnConfig epilog which write the model import and task functions import
+including serializing their parameters.
+"""
+from typing import Any, Dict, Optional
+
+from i6_experiments.common.setups.returnn_pytorch.serialization import Collection
+from i6_experiments.common.setups.serialization import ExternalImport, Import, PartialImport
+
+from . import PACKAGE
+from .default_tools import I6_MODELS_REPO_PATH, FAIRSEQ_PATH
+
+
+def serialize_training(
+    network_module: str,
+    net_args: Dict[str, Any],
+    unhashed_net_args: Optional[Dict[str, Any]] = None,
+    debug: bool = False,
+) -> Collection:
+    """
+    Helper function to create the serialization collection
+
+    :param network_module: path to the pytorch config file containing Model
+    :param net_args: arguments for the model
+    :param unhashed_net_args: as above but not hashed
+    :param debug: run training in debug mode: linking from recipe instead of copy
+    :return: Collection object to be added to the ReturnnConfig epilog
+    """
+    package = PACKAGE + ".pytorch_networks"
+
+    pytorch_model_import = PartialImport(
+        code_object_path=package + ".%s.Model" % network_module,
+        unhashed_package_root=PACKAGE,
+        hashed_arguments=net_args,
+        unhashed_arguments=unhashed_net_args or {},
+        import_as="get_model",
+    )
+    pytorch_train_step = Import(
+        code_object_path=package + ".%s.train_step" % network_module, unhashed_package_root=PACKAGE
+    )
+    i6_models = ExternalImport(import_path=I6_MODELS_REPO_PATH)
+    fairseq = ExternalImport(import_path=FAIRSEQ_PATH)
```

Comment:
I wonder whether there is a more flexible way to add this to the config if we might need to switch between fairseq versions for different fine-tunings.

Reply:
I think `default_tools.py` is the right place to store this fairseq path, as it can be accessed globally that way. Maybe we could add a `set_fairseq_path` function in that file, that can be called from our function in `baseline.py` to use a specified fairseq version. What do you think about that?

Reply:
Yes we can do that. But don't forget to serialize it for the config.

Reply:
I implemented that with the newest commit (b4744ce)